library/ssl_tls13_generic.c

```diff
@@ -273,10 +277,85 @@ static int ssl_tls13_parse_certificate_verify( mbedtls_ssl_context *ssl,
         goto error;
     }
 
-    if( mbedtls_ssl_tls13_get_pk_type_and_md_alg_from_sig_alg(
-                                        algorithm, &sig_alg, &md_alg ) != 0 )
+    switch( algorithm )
```

Comment:
Hmm, now that I think about it, it might be cleaner to do this by calling `mbedtls_ssl_tls13_get_pk_type_and_md_alg_from_sig_alg()` followed by `mbedtls_psa_translate_md()`.

Reply:
If we are moving away from md, and we want to use psa API, we should limit the use of md. This seems to me like a step back. If we want cleaner code, I can move this switch to mbedtls_ssl_tls13_get_pk_type_and_psa_alg_from_sig_alg() I can create. I see no point in going through mbedtls_md_type_t. What do you think?

Reply:
Good point, thanks for considering that! There are roughly 3 kinds of possible uses of MD:

1. Using its functions for cryptographic computations. We are absolutely moving away from that right now, as a matter of fact this step will be fully complete one this PR (including the extra bit I requested late) is merged (so, you get the honour of getting us over the line). (For layers other than MD, it's already complete except for FFDH, EC J-PAKE and some cases of RSA-PSS verification.) Once this is done, in a PSA-enabled build, all crypto computations go through PSA, so accelerators when available.
2. Using its functions for something else, such as translating from identifiers to sizes or even human-readable strings. Moving away from that was not a goal until recently, but for MD it's almost done, only a few cases remain (#5795 and #5796). Once that's done, we can start working on PSA-enabled builds that don't need `md.c` (but still possibly use `md.h`) in order to save code size - though it's not entirely trivial due to deep dependencies, see #5210.
3. Using its symbols other that functions (types, enum values, etc. - things that only need `md.h` but not `md.c`). We'll be moving away from that at one point, but this can't be done fully until the next major version of Mbed TLS, as types from `md.h` are embedded in the public API of other modules. Also, while not morally satisfying, it has little practical consequence other than the need for translation functions like `mbedtls_psa_translate_md()`. So, it's not really a target right now.

That said, just because it's not a target doesn't prevent us from making improvements in that direction when we notice an opportunity :) In that case, the thing I'd like to avoid is duplication between `mbedtls_ssl_tls13_get_pk_type_and_md_alg_from_sig_alg()` and `mbedtls_ssl_tls13_get_pk_type_and_psa_alg_from_sig_alg()` (or currently, the embedded switch statement). So if we have two functions, one returning an `md_alg` and the other `psa_alg`, we should make sure one is defined in terms of the other. I basically suggesting basing the `psa_alg` version on the `md_alg` one by using `mbedtls_psa_translate_md()` but we could very well do it this other way, using `mbedtls_md_info_from_psa()`.

Either way, moving away from using symbols from `md.h` internally is not a goal right now, but you can still do it opportunistically if you want (when it's small enough, like here). The only thing I really care about in this case is to avoid duplication.